etc/spack/defaults/config.yaml

```diff
@@ -16,28 +16,32 @@
 config:
   # This is the path to the root of the Spack install tree.
   # You can use $spack here to refer to the root of the spack instance.
-  install_tree:
-    root: $spack/opt/spack
-    projections:
-      all: "${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}"
-    # install_tree can include an optional padded length (int or boolean)
+  install_trees:
+    default:
```

Comment:
I think the term "default" is too generic: from the user's perspective the default install tree should change based on configuration of shared install trees: if there are any, then it should be using `~`; otherwise it should be using `$spack`.

On a side note, in my original approach I generated these on the fly, since I think generally speaking only one of these install trees will ever be used; I need to think on whether it's confusing to have both of them live in the default configuration.

Reply:
Changed to `spack-root` instead. Its possible that there is a better name, but this removes the ambiguity of `default` which could change.